cmd/prometheus/main.go

```diff
@@ -186,6 +202,12 @@ func reloadConfig(filename string, rls ...Reloadable) bool {
 	for _, rl := range rls {
 		success = success && rl.ApplyConfig(conf)
 	}
+	if success {
+		configSuccess.Set(1.0)
+		configSuccessTime.Set(float64(time.Now().Unix()))
+	} else {
+		configSuccess.Set(0)
```

Comment:
If you want to get rid of the multiple places where you have to remember to set the metric to the right value, the best would be to use a named return value:

```go
func reloadConfig(filename string, rls ...Reloadable) (success bool) {
```

...and then do this at the beginning of the function:

```go
defer func() {
  if success {
    configSuccess.Set(1)
    configSuccessTime.Set(float64(time.Now().Unix()))
  } else {
    configSuccess.Set(0)
  }
}()
```